torch/_higher_order_ops/flex_attention.py

```diff
@@ -473,6 +497,27 @@ def flex_attention_fake_tensor_mode(
     score_mod_other_buffers: tuple = (),
     mask_mod_other_buffers: tuple = (),
 ) -> tuple[torch.Tensor, torch.Tensor]:
+    flat_args, _ = pytree.tree_flatten(
```

Comment:
FunctionalTensor and FakeTensor implementations should just be routed to the torch_dispatch. We've got a way of routing the FakeTensor implementation already:  https://github.com/pytorch/pytorch/blob/159e2f96e3d7aec0187b03c483e177c5e96b1ebb/torch/_higher_order_ops/utils.py#L799-L804

Could you try that for the FakeTensorMode registration?

Reply:
will try!